artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -324,13 +325,7 @@ public String debug() {
          out.println("consumer: " + holder.consumer.debug());
       }
 
-      for (MessageReference reference : intermediateMessageReferences) {
-         out.print("Intermediate reference:" + reference);
-      }
-
-      if (intermediateMessageReferences.isEmpty()) {
-         out.println("No intermediate references");
-      }
+      out.println("Intermediate reference size is " + intermediateMessageReferences.size());
```

Comment:
Was this change needed, seem this is some existing debug thats been useful historically to debug issues, any harm leaving as is?

Reply:
MpscUnboundedArrayQueue doesn't support iterator, so I had to remove it.

Reply:
I will take a note  to implement one on JCTools :+1:

Reply:
Do we need MpscUnboundedArrayQueue then in this PR to make it work? Im conscious of removing useful debugging features. Perf shouldnt be at the cost of feature / operational need.

Can always be re-added back once it supports? In a later PR perf improvement.

Reply:
@franz1981 is this relatively trivial? Could always copy the class in temporarily and add the feature in artemis copy as an interim. (also would avoid a karaf issue with extra deps)

Reply:
Not so easy I think...for bounded versions of qs yet (already supported in the current/maybe next release), but for unbounded is not that easy...
We are sure we really use iterator? or is iterator + clear at the end?

Reply:
So debug like this only make it into code if it was needed at some point to debug an operational issue, as such i would suggest it was needed. My question is how much perf improvement is actually gained from this one particular change. I would expect the real perf was in the change in paging logic not the change of queue.

Reply:
MpscUnboundedArrayQueue might perform well in journal mode where there are many producers and one consumer. The main reason I used MpscUnboundedArrayQueue is I need to calculate size of messages both in intermediateMessageReferences and messageReferences, i.e.`intermediateMessageReferences.size() + messageReferences.size() < MAX_DEPAGE_NUM` in QueueImpl::needsDepage. However time complexity of size() for ConcurrentLinkedQueue is O(n), so I replaced it with MpscUnboundedArrayQueue(I did't find other queues fit for our scenario). I didn't think about issues like kafraf, debug by introducing this.